src/providers/SubscriptionProvider/SubscriptionProvider.container.js

```diff
@@ -6,98 +6,33 @@ import API from '../../api';
 import { isAndroid } from '../../cordova-util';
 
 import {
-  updateProductState,
+  updateAndroidSubscriptionState,
   updateIsSubscribed,
-  updateSubscription
+  updateSubscription,
+  comprobeSubscription
 } from './SubscriptionProvider.actions';
 import { getProductStatus } from '../../components/Settings/Subscribe/Subscribe.helpers';
 import { onAndroidResume } from '../../cordova-util';
+import { NOT_SUBSCRIBED, PROCCESING } from './SubscriptionProvider.constants';
 
 export class SubscriptionProvider extends Component {
   static propTypes = {
     children: PropTypes.node.isRequired
   };
 
   componentDidMount() {
-    const { isSubscribed } = this.props;
+    const { isSubscribed, comprobeSubscription } = this.props;
     if (isAndroid()) {
       this.configInAppPurchasePlugin();
       if (isSubscribed) {
-        this.comprobeSubscription();
+        comprobeSubscription();
       }
-      onAndroidResume(() => this.comprobeSubscription());
+      onAndroidResume(() => comprobeSubscription());
```

Comment:
consider the case that the user Paid a subscription on another device. Create an issue